src/clj_irods/core.clj

```diff
@@ -153,52 +166,58 @@
   [irods user zone path]
-  (from-stat-or-listing :date-created (comp (partial * 1000) #(Integer/parseInt %) :create_ts) irods user zone path))
+  (otel/with-span [s ["date-created"]]
+    (from-stat-or-listing :date-created (comp (partial * 1000) #(Integer/parseInt %) :create_ts) irods user zone path)))
 
 (defn file-size
   "The number of bytes of a path. 0 for folders."
   [irods user zone path]
-  (from-stat-or-listing :file-size :data_size irods user zone path))
+  (otel/with-span [s ["file-size"]]
+    (from-stat-or-listing :file-size :data_size irods user zone path)))
 
 (defn uuid
   "Get the UUID (via the ipc_UUID AVU) for a path."
   [irods user zone path]
-  (cached-or-get irods
-    [from-listing :uuid user zone path]
-    [(fn [cache? irods path]
-       (let [get-from-metadata (fn [metadata]
-                                 (when-let [uuid-meta (first (filter #(= (:attr %) "ipc_UUID") @metadata))]
-                                   (delay (get uuid-meta :value))))]
-           (when-let [metadata (if cache?
-                                 (jargon/cached-get-metadata irods path)
-                                 (jargon/get-metadata irods path :known-type @(object-type irods user zone path)))]
-             (or
-               (get-from-metadata metadata)
-               nil))))
-     path]))
+  (otel/with-span [s ["uuid"]]
+    (cached-or-get irods
+      [from-listing :uuid user zone path]
+      [(fn [cache? irods path]
+         (let [get-from-metadata (fn [metadata]
+                                   (when-let [uuid-meta (first (filter #(= (:attr %) "ipc_UUID") @metadata))]
+                                     (instrumented-delay (get uuid-meta :value))))]
+             (when-let [metadata (if cache?
+                                   (jargon/cached-get-metadata irods path)
+                                   (jargon/get-metadata irods path :known-type @(object-type irods user zone path)))]
+               (or
+                 (get-from-metadata metadata)
+                 nil))))
+       path])))
 
 (defn permission
   "Permission for a user on a path. Returned as a keyword like :read :write :own, or nil"
   [irods user zone path]
-  (cached-or-get irods
-    [from-listing (comp jargon-perms/fmt-perm :access_type_id) user zone path]
-    [(fn [cache? irods user path]
-       (if cache?
-         (jargon/cached-permission-for irods user path)
-         (delay
-           @(jargon/permission-for irods user path :known-type @(object-type irods user zone path)))))
-     user path]))
+  (otel/with-span [s ["permission"]]
+    (cached-or-get irods
+      [from-listing (comp jargon-perms/fmt-perm :access_type_id) user zone path]
+      [(fn [cache? irods user path]
+         (if cache?
+           (jargon/cached-permission-for irods user path)
+           (instrumented-delay
+             @(jargon/permission-for irods user path :known-type @(object-type irods user zone path)))))
```

Comment:
(note to self, probably should do the thing to ensure calculation starts immediately here too)